### PR TITLE
fix: Search term should not get added to event categories

### LINKED ISSUE #6034: Frontpage Search/Event Category: Search term should not get added to event categories
There are event categories/tags available on the bottom of each event card. If the user has done a search on the system before, the search term is always added as "filter". Tags on event categories should always link to the same standard page and they should not have a filter added.

Screenshot from 2020-12-15 10-43-12

### PR DESCRIPTION
**Pull request by sachinchauhan2889**
Fixes #6034 

#### Checklist

- [X] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [X] My branch is up-to-date with the Upstream `development` branch.
- [X] The acceptance, integration, unit tests and linter pass locally with my changes 
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/routes/explore.js [hunk 1]
```diff
@@ -249,4 +249,9 @@ export default class ExploreRoute extends Route {
   queryParamsDidChange(change, params) {
     this.debouncedFilterChange(params);
   }
+
+  resetController(controller) {
+    super.resetController(...arguments);
+    controller.clearAllFilters();
```
Comment (daretobedifferent18) [inline app/routes/explore.js:255]: Why not using clearFilter(), this results in duplication of code

Comment (sachinchauhan2889) [reply app/routes/explore.js:255]: clear filter clears only one filter at a time. at resetController i need to call it for every filterOptions.

File: app/controllers/explore.js [hunk 2]
```diff
@@ -67,4 +67,21 @@ export default class ExploreController extends Controller {
       this.set('cfs', null);
     }
   }
+
+  @action
+  clearAllFilters() {
+    this.set('event_name', null);
```
Comment (daretobedifferent18) [inline app/controllers/explore.js:73]: use instead ```this.setProperties({})```

**Review by daretobedifferent18 — COMMENTED** [review]
There is a function clearFilter already present in sidebar.js file for explore/sidebar.
So inorder to reduce the duplication pls remove that function from there and pass the action u defined now there to use the same.

File: app/controllers/explore.js (hunk 2 above)
Comment (sachinchauhan2889) [reply app/controllers/explore.js:73]: > use instead `this.setProperties({})`

any specific reason for using this.setProperties({})

Comment (daretobedifferent18) [reply app/controllers/explore.js:73]: sometimes when ```this.set``` is called multiple times result in producing minor bugs

**Comment by sachinchauhan2889** [comment]
@daretobedifferent18 

https://github.com/fossasia/open-event-frontend/pull/6007#pullrequestreview-551728246

I was telling about github actions / lint annotations. 

After merging that PR https://github.com/fossasia/open-event-frontend/pull/6007 . Github actions is showing that annotaions in every new PR.  you can check in my PR also.

**Comment by daretobedifferent18** [comment]
> @daretobedifferent18
> 
> [#6007 (review)](https://github.com/fossasia/open-event-frontend/pull/6007#pullrequestreview-551728246)
> 
> I was telling about github actions / lint annotations.
> 
> After merging that PR #6007 . Github actions is showing that annotaions in every new PR. you can check in my PR also.

Yup I get to know, I will be fixing it in #6017

**Comment by mariobehling** [comment]
As the search feature has been updated in recent PRs I am updating this branch. Thank you.

**Comment by mariobehling** [comment]
Thank you. The issue also appears on the explore/search page itself. The categories of the cards get the search key added. I think in the cards itself there should not be a filter added as this is contrary to the experience in other search engines.
Compare https://open-event-frontend-git-resetcontroller-at-explore.eventyay.vercel.app/explore?event_name=summit

**Comment by sachinchauhan2889** [comment]
@iamareebjamal sir, please review

**Review by maze-runnar — COMMENTED** [review]
working for me.

**Pull request merged**